### PR TITLE
Cross encoding

### PR DESCRIPTION
**Pull request by jeevanp03**
Co-authored-by: Helen @Helenessli 

Integrated a cross-encoding re-ranker in both the vanilla RAG and RAPTOR flows and updated the README.md in the RAG folder.

### CODE REVIEW

**Review by hargunmujral — APPROVED** [review]
Lgtm, did you verify it integrates end to end with the frontend?

File: RAG/chroma_embedding.py [hunk 1]
```diff
@@ -166,13 +163,22 @@ def get_similar_documents(self, query) -> list[tuple[float, int, str]]:
...
-                (0.0, doc.metadata['seq_num'], doc.page_content))
-        return parsed_docs
+        if rerank:
+            print("Reranking documents...")
+            return self.__rerank(query_text, top_n=top_n,
+                                 search_kwargs=search_kwargs)
+
+        return self.__chroma_db.similarity_search(query_text)
+
+    def __rerank(self, prompt, top_n=3, search_kwargs=20):
```
Comment (hargunmujral) [inline RAG/chroma_embedding.py:173]: Why top 3? Shouldn't we do more

**Comment by jeevanp03** [comment]
> Lgtm, did you verify it integrates end to end with the frontend?

Yes

File: RAG/chroma_embedding.py (hunk 1 above)
Comment (jeevanp03) [reply RAG/chroma_embedding.py:173]: Cause of context window right. I would rather have top 3. U have seen the needle in the haystack metric right

**Comment by hargunmujral** [comment]
We should also still verify that the quality of raptor retrieval is tangibly better than normal rag. I imagine the way we chunk still has a large impact on quality of retrieval.

File: RAG/chroma_embedding.py (hunk 1 above)
Comment (hargunmujral) [reply RAG/chroma_embedding.py:173]: Can you elaborate? Also if we switch to 4o which has a longer context window, we should def pass in more context. 

The reason I think more context is better in this case is that we aren't passing in few shot examples, instead we are trying to pass in more medical info that can be useful for the rag response

**Pull request merged**